src/lib.rs

```diff
@@ -211,7 +324,7 @@ impl Parser {
 
         for i in 0..self.osc_num_params {
             let indices = self.osc_params[i];
-            slices[i] = &self.osc_raw[indices.0..indices.1];
+            slices[i] = &self.osc_raw.as_slice()[indices.0..indices.1];
```

Comment:
This `as_slice` is unnecessary, Vecs will be automatically sliced to slices.

Reply:
`as_slice` is necessary because `nostd` doesn't have `Vec` and so `OscBuffer` doesn't automatically slice in that way; removing `as_slice` causes the tests to fail in the `no_std` case.